app/models/user.rb

```diff
@@ -524,10 +526,11 @@ def temp_name_exists?
   end
 
   def temp_username
-    if twitter_username
-      twitter_username.downcase.gsub(/[^0-9a-z_]/i, "").delete(" ")
-    elsif github_username
-      github_username.downcase.gsub(/[^0-9a-z_]/i, "").delete(" ")
+    Authentication::Providers.username_fields.each do |username_field|
```

Comment:
This is a very large PR, it might be nice to pull out a bunch of these refactors into their own PR so this is not as big and we can push them out and ensure everything works as refactored before trying to add an additional auth provider.

Reply:
After a quick conversation with Josh we both agree that this seems like a very good idea. We still haven't been able to fix this CSRF issue despite lots of debugging and I agree having a smaller PR could help making it easier to receive feedback/help.

I'm hoping to split this one today into a new PR with the generalization refactor and this PR will remain as "Sign in with Apple" PR :)